src/strands/agent/agent.py

```diff
@@ -365,6 +365,16 @@ def __init__(
                 self.hooks.add_hook(hook)
         self.hooks.invoke_callbacks(AgentInitializedEvent(agent=self))
 
+    @property
+    def system_prompt(self) -> str | None:
```

Comment:
Can we add some docs indicating the need/purpose of the getters.

From a reader perspective (not looking at the PR) it's not at all obvious why the system_prompt returns a string, but the system_prompt_content accepts non-strings